File: src/main/java/com/lowtuna/jsonblob/view/ApiView.java
package com.lowtuna.jsonblob.view;

import com.lowtuna.dropwizard.extras.config.GoogleAnalyticsConfig;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Set;

@Data
@EqualsAndHashCode(callSuper = true)
public class ApiView extends JsonBlobView {

    public ApiView(String gaWebPropertyID, String pageName, Set<GoogleAnalyticsConfig.CustomTrackingCode> customTrackingCodes) {
        super("/api.hbs", gaWebPropertyID, pageName, customTrackingCodes);
    }

}

File: src/main/kotlin/jsonblob/api/http/view/ApiView.kt
package jsonblob.api.http.view

import io.micronaut.core.annotation.Introspected
import jsonblob.config.GoogleAnalyticsConfig

@Introspected
class ApiView(
    gaWebPropertyID: String,
    pageName: String,
    gaCustomTrackingCodes: Set<GoogleAnalyticsConfig.CustomTrackingCode>
) : JsonBlobView(gaWebPropertyID, pageName, gaCustomTrackingCodes)